### PR TITLE
Add user sign out

### PR DESCRIPTION
**Pull request by romiparada**
https://trello.com/c/5s37UrqD

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+JWT_SECRET=secret
```
Comment (nerlichman) [inline .env:1]: We shouldn't push the `.env` file to GitHub, you could add it to `.gitignore` to prevent it.

To showcase the env variables needed for the project we typically use the `.env.sample` file

File: spec/requests/api/users/sessions/destroy_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'DELETE /api/users/sign_out', type: :request do
+  subject { delete destroy_user_session_path, params: {}, headers:, as: :json }
+
+  let(:user) { create(:user) }
+  let(:headers) { Devise::JWT::TestHelpers.auth_headers({}, user) }
```
Comment (juan-apa) [inline spec/requests/api/users/sessions/destroy_spec.rb:9]: let's wrap this Devise::JWT::TestHelpers.auth_headers in a test helper. e.g.:
```ruby
module AuthHelper
  def get_jwt(user)
    auth_headers = Devise::JWT::TestHelpers.auth_headers({}, user)
    auth_headers['Authorization']
  end

  def auth_headers(user)
    { 'Authorization' => get_jwt(user) }
  end
end
```

File: spec/requests/api/users/sessions/destroy_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    it 'revokes the user jwt token' do
+      expect { subject }.to(change { user.reload.jti })
+    end
+  end
+
+  context 'when the credentials are incorrect' do
+    context 'when the auth header is not set' do
+      let(:headers) { {} }
+      it 'returns 204 status code' do
```
Comment (juan-apa) [inline spec/requests/api/users/sessions/destroy_spec.rb:25]: add an empty line between `let` and `it`

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
+  include ActionController::MimeResponds
```
Comment (juan-apa) [inline app/controllers/application_controller.rb:4]: why is this necessary?

Comment (romiparada) [reply app/controllers/application_controller.rb:4]: if I comment out that line I get the following error: 
```
  1) DELETE /api/users/sign_out when the credentials are correct returns 204 status code
     Failure/Error: subject { delete destroy_user_session_path, params: {}, headers:, as: :json }

     NoMethodError:
       undefined method `respond_to' for #<Api::Users::SessionsController:0x0000000000d980>
     # ./spec/requests/api/users/sessions/destroy_spec.rb:6:in `block (2 levels) in <main>'
     # ./spec/requests/api/users/sessions/destroy_spec.rb:13:in `block (3 levels) in <main>'
```

**Review by juan-apa — APPROVED** [review]
